File: BackEnd/src/main/java/com/example/futurbe/entitys/Application.java
package com.example.futurbe.entitys;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.Setter;

import java.util.Date;

@Entity
@AllArgsConstructor
@Getter
@Setter
public class Application {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @ManyToOne
    @JoinColumn(name = "offer_id", nullable = false)
    private Offre offer;

    @ManyToOne
    @JoinColumn(name = "user_id", nullable = false)
    private User user;

    private Date applicationDate;
    private boolean is_viewed ;

    public Application() {

    }

    // Getters and setters
}

File: FrontEnd/src/app/models/Application.ts
import { Offer } from "./offer";
import { UserDTO } from "./user.dto";


export interface Application {
    id: number;
    applicationDate : string;
    offerId: number;
    userId: number;
    is_viewed: boolean;
    offerDetails ?: Offer;
    userDetails ?: UserDTO

  }